client/src/components/GameContainer.js

```diff
@@ -62,6 +58,19 @@ export default class GameContainer extends Component {
     });
   }
 
+  componentWillUnmount() {
+    this.disconnect();
+  }
+
+  connect = () => {
```

Comment:
This method could be a little clearer, maybe `connectToSocket`? It wasn't immediately obvious what was happening on line 20

Reply:
Also why isn't the socket set on the state of the component?